src/decoder.rs

```diff
@@ -1047,20 +1053,12 @@ mod tests {
             }
         );
 
-        // Note: For compressed instructions, the immediate value (imm) represents an offset
-        // and is multiplied by the necessary factor (e.g., x4/8) during decoding.
-        // Therefore, we actually decode an offset as the immediate value for compressed instructions.
-
-        // Example:  0xffff4798 would decode into c.lw x14, 8(x15) (immediate value is 8),
-        // but we assert imm as 32 (8*4), as the eventual offset value would be 32.
-        // This helps to keep compressed and uncompressed instructions handlers more homogenous in other modules.
-
         assert_eq!(
             mctx.decode_store(0xffffe798),
             Instr::Store {
                 rs2: Register::X14,
                 rs1: Register::X15,
-                imm: 64,
+                imm: 8,
```

Comment:
In short what was the issue @francois141 ? Why shouldn't we multiple the immediate to get the final offset?